### PR TITLE
HDDS-7641. Namespace quota validation is not present in multiple places

### PR DESCRIPTION
**Pull request by sumitagrawl**
## What changes were proposed in this pull request?

1. missing validation for namespace added
2. missing increment of namespace added where parent directory created (as missing)
3. Cache update for directory done after validation of quota

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-7641

## How was this patch tested?

E2E perspective for additional scenario, this is tested

### CODE REVIEW

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/key/OMKeyCreateResponse.java [hunk 1]
```diff
@@ -91,6 +91,13 @@ public void addToDBBatch(OMMetadataManager omMetadataManager,
         omMetadataManager.getKeyTable(getBucketLayout())
             .putWithBatch(batchOperation, parentKey, parentKeyInfo);
       }
+      
+      // namespace quota changes for parent directory
+      String bucketKey = omMetadataManager.getBucketKey(
+          getOmBucketInfo().getVolumeName(),
+          getOmBucketInfo().getBucketName());
+      omMetadataManager.getBucketTable().putWithBatch(batchOperation,
```
Comment (sadanand48) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/key/OMKeyCreateResponse.java:99]: Looks like the usedBytes/usedNamespace  update during key/file create was intentionally removed as part of [HDDS-6556](https://issues.apache.org/jira/browse/HDDS-6556) and the update is only upon Key commit. We should perhaps remove this.
cc @captainzmc @errose28

Comment (sumitagrawl) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/key/OMKeyCreateResponse.java:99]: @sadanand48 Here we are updating namespace of parent directory of key only, still usedBytes and namespace of the key is done at commitKey only.

But during prepare, we just do validation if quota as required is valid or not which is already present.

Comment (sadanand48) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/key/OMKeyCreateResponse.java:99]: Got it , thanks!

**Review by errose28 — COMMENTED** [review]
Thanks for the fix @sumitagrawl. Can we add some unit tests for these cases?

**Comment by sumitagrawl** [comment]
> Thanks for the fix @sumitagrawl. Can we add some unit tests for these cases?

@errose28 updated test cases for all scenarios

**Review by errose28 — APPROVED** [review]
Thanks for adding the tests @sumitagrawl. This looks good for bucket quota. I think volume quota is still not fully implemented so we can deal with that in follow up tasks if necessary.

**Pull request merged**